mbuild/compound.py

```diff
@@ -348,6 +561,15 @@ def add(self, new_child, label=None, containment=True, replace=False,
                              'can be added to Compounds. You tried to add '
                              '"{}".'.format(new_child))
 
+        if not new_child.port_particle:
+            # If we are adding a Compound containing rigid particles
+            if new_child.contains_rigid:
+                # If we are adding to a Compound already containing rigid particles
+                if self.contains_rigid:
+                    # If we want to reset the rigid_ids
```

Comment:
I think now with the refactored API these comments are superfluous since the code speaks quite clearly for itself.